bogus msvc ci builds for arm64

### Sonivox version
Commit 2c16308

### Describe the bug
The Windows CI job for ARM64 https://github.com/pedrolcl/sonivox/actions/runs/14827987278/job/41624014851 builds using the x64 host and x64 target, instead of building for arm64

when building for arm64, the link phase fails with messages like:
```
sonivox-static.lib(eas_smf.c.obj) : error LNK2019: unresolved external symbol _umul128 referenced in function compute_tick_conv
sonivox-static.lib(eas_smf.c.obj) : error LNK2019: unresolved external symbol _addcarry_u64 referenced in function compute_tick_conv
```

The explanation is that both _umul128 and _addcarry_u64 intrinsics are defined only for Intel x64 builds.

### Expected behavior

the CI job should specify an arm64 target. And alternative functions for ARM64 in order to properly link are needed.

#### Additional context

There is an open issue on ilammy/msvc-dev-cmd about arm64 native support:

https://github.com/ilammy/msvc-dev-cmd/issues/90

--- a/.github/workflows/win-msvc.yml
+++ b/.github/workflows/win-msvc.yml
@@ -27,6 +27,8 @@ jobs:
 
     - name: '${{ matrix.icon }} Setup MSVC'
       uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ matrix.arch }}
 
     - name: '${{ matrix.icon }} Configure CMake'
       run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_INSTALL_PREFIX=${{env.INSTALL_LOCATION}} -G Ninja

--- a/arm-wt-22k/lib_src/compute_tick_conv.c
+++ b/arm-wt-22k/lib_src/compute_tick_conv.c
@@ -2,6 +2,19 @@
 #include <limits.h> // For USHRT_MAX
 #ifdef _MSC_VER
 #include <intrin.h> // For _umul128 and _addcarry_u64
+#if defined(_M_ARM64) //!(defined(_M_AMD64) || defined(_M_X64))
+#pragma intrinsic(__umulh)
+
+uint64_t _umul128(uint64_t a, uint64_t b, uint64_t *high) {
+    *high = __umulh(a, b);
+    return a * b;
+}
+
+uint8_t _addcarry_u64(uint64_t carry_in, uint64_t x, uint64_t y, uint64_t *sum) {
+    *sum =  x + y + (carry_in !=0 ? 1 : 0);
+    return x > UINT32_MAX - y;
+}
+#endif
 #endif
 
 #define MAX_TICK_CONV USHRT_MAX // Use USHRT_MAX for 16-bit unsigned max value

